js_modules/dagster-ui/packages/ui-core/src/nav/LeftNavItem.tsx

```diff
@@ -21,8 +27,31 @@ export const LeftNavItem = React.forwardRef(
     const {label, leftIcon, path, repoAddress, schedules, sensors} = item;
 
     const [showDialog, setShowDialog] = React.useState(false);
+    const repositoryId = useRepository(repoAddress)?.repository.id;
 
-    const rightIcon = () => {
+    useQuery<InstigationStatesQuery, InstigationStatesQueryVariables>(INSTIGATION_STATES_QUERY, {
+      variables: {
+        repositoryID: repositoryId!,
+      },
+      skip: !repositoryId,
+    });
+    const [data, setData] = useState<Record<string, InstigationStateFragment>>({});
+
+    const status = useMemo(() => {
+      const anyScheduleIsRunning = schedules.some((schedule) => {
+        const state = data[schedule.id];
+        return (state?.status ?? schedule.scheduleState.status) === InstigationStatus.RUNNING;
+      });
+      const anySensorIsRunning = sensors.some((sensor) => {
+        const state = data[sensor.id];
+        return (state?.status ?? sensor.sensorState.status) === InstigationStatus.RUNNING;
+      });
+      return anyScheduleIsRunning || anySensorIsRunning
+        ? InstigationStatus.RUNNING
+        : InstigationStatus.STOPPED;
+    }, [schedules, sensors, data]);
+
+    const rightIcon = useMemo(() => {
```

Comment:
It feels like rightIcon should also be based on whether or not they are present AND running? eg: in this case I can see status = Running because of a sensor, and the schedule icon because we prefer those, even if the schedule is off.

Pretty tangential to this PR though no need to chnage